http/test/test_client.lua

```diff
@@ -27,6 +27,47 @@ if not(resp.code == 200) then error("resp code") end
 if not(resp.body == "OK") then error("resp body") end
 print("done: http.client:getBasicAuth")
 
+local req, err = http.file_request("http://127.0.0.1:1111/upload", {
+  fieldname = "file",
+  path = "./test/data/test.txt"
+})
+if err then error(err) end
```

Comment:
**suggestion; not-blocking**: I guess this is the prevailing style… I really would rather see use of assert… it's so much more compact and also follows style - style would have this be split into 

```lua
if err then
    error(err)
end
```
which is pretty "big" vertically.

Instead… this could be very concisely this (as could any of the checks with error)

```suggestion
assert(not err, err)
```